qa/rpc-tests/mempool_push.py

```diff
@@ -0,0 +1,163 @@
+        # kick us out of IBD mode since the cached blocks will be old time so it'll look like our blockchain isn't up to date
+        # if we are in IBD mode, we don't request incoming tx.
+        self.nodes[0].generate(1)
+
+        logging.info("ancestor count test")
+        bal = self.nodes[1].getbalance()
+        addr = self.nodes[1].getnewaddress()
+
+        txhex = []
+        for i in range(0,BCH_UNCONF_DEPTH*2):
+          try:
+            txhex.append(self.nodes[1].sendtoaddress(addr, bal-1))  # enough so that it uses all UTXO, but has fee left over
+          except JSONRPCException as e: # an exception you don't catch is a testing error
+              print(str(e))
+              raise
+        assert_equal(self.nodes[0].getmempoolinfo()["size"], BCH_UNCONF_DEPTH)
+        assert_equal(self.nodes[1].getmempoolinfo()["size"], BCH_UNCONF_DEPTH*2)
+        # Set small to commit just a few tx so we can see if the missing ones get pushed
+        self.nodes[0].set("mining.blockSize=6000")
+        blk = self.nodes[0].generate(1)[0]
+        blkhex = self.nodes[0].getblock(blk)
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+        # generate the block somewhere else and see if the tx get pushed
+        self.nodes[2].set("mining.blockSize=4000")
+        blk2 = self.nodes[2].generate(1)[0]
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == BCH_UNCONF_DEPTH)
+
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getbestblockhash() == blk2)  # make sure its settled so we can get a good leftover count for the next test.
+        unconfLeftOver = self.nodes[1].getmempoolinfo()["size"]
+        assert(unconfLeftOver >= BCH_UNCONF_DEPTH)  # if someone bumps the BCH network unconfirmed depth, you need to build a bigger unconf chain
+        # Let's consume all BCH_UNCONF_DEPTH tx
+        self.nodes[0].set("mining.blockSize=8000000")
+
+        waitFor(DELAY_TIME, lambda: len(self.nodes[0].getblocktemplate()["transactions"])>=BCH_UNCONF_DEPTH)
+        blk3 = self.nodes[0].generate(1)[0]
+        blk3data = self.nodes[0].getblock(blk3)
+        # this would be ideal, but a particular block is not guaranteed to contain all tx in the mempool 
+        # assert_equal(len(blk3data["tx"]), BCH_UNCONF_DEPTH + 1)  # chain of BCH_UNCONF_DEPTH unconfirmed + coinbase
+        committedTxCount = len(blk3data["tx"])-1  # -1 to remove coinbase
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getbestblockhash() == blk3)
+        waitFor(DELAY_TIME, lambda: self.nodes[1].getmempoolinfo()["size"] == unconfLeftOver - committedTxCount)
+        # make sure that everything that can be pushed is pushed
+        waitFor(DELAY_TIME, lambda: self.nodes[0].getmempoolinfo()["size"] == min(unconfLeftOver - committedTxCount, BCH_UNCONF_DEPTH))
+
+        # clean up: confirm all the left over tx from the prior test
+        self.nodes[0].set("mining.blockSize=8000000")
```

Comment:
the above is not necessary you already set node 0 `mining.blockSize` to 8MB on line 78.